R/bootstrap.nlsfit.R

```diff
@@ -113,55 +60,146 @@ bootstrap.nlsfit <- function(fn,
     if(boot.R != dbs[1]) {
       stop("boot.R inconsistent with dimension one of bsamples!")
     }
-    ## add original data as first row
+	## add original data as first row
     bsamples <- rbind(Y, bsamples)
   }
+  if(!useCov){
+    if(missing(dy)){
+      dY <- apply(X=bsamples, MARGIN=2, FUN=sd)
+      dy <- dY[1:(length(y))]
+      if(errormodel == "xyerrors"){
+        dx <- dY[(length(y)+1):length(Y)]
+      }
+    }else{
+      ## cast dy to dY
+      if(errormodel == "xyerrors") {
+        dY <- c(dy, dx)
+      }else{
+        dY <- dy
+      }
+    }
+  }
 
   ## generate bootstrap samples if needed
   ## and invert covariance matrix, if applicable
-  if(sim == "parametric") {
-    for(i in seq_along(Y)) {
-      bsamples[rr, i] <- rnorm(n=boot.R, mean = Y[i], sd = dY[i])
+  if(useCov) {
+    inversion.worked <- function(InvCovMatrix) {
+      if(inherits(InvCovMatrix, "try-error")) {
+        stop("Variance-covariance matrix could not be inverted!")
+      }
     }
-    dY <- diag(1./dY)
-  }
-  else if(useCov) {
-    CovMatrix <- cov(bsamples)
-    InvCovMatrix <- try(invertCovMatrix(bsamples, boot.l=1, boot.samples=TRUE), silent=TRUE)
-    if(inherits(InvCovMatrix, "try-error")) {
-      stop("Variance-covariance matrix could not be inverted!")
+    if(missing(CovMatrix)){
+      InvCovMatrix <- try(invertCovMatrix(bsamples, boot.l=1, boot.samples=TRUE), silent=TRUE)
+      inversion.worked(InvCovMatrix)
+      dY <- chol(InvCovMatrix)
```

Comment:
after our discussion, why do you need the Cholesky decomposition here?

Reply:
I don't really need it here.
But I need it for parametric bootstrap with covariance. And I need it in case the gradient is used.
So I simply left it here for consistency. I am aware that this is not the most performant way, but the difference should not be significant because there is the inversion anyway.